src/parser.cpp

```diff
@@ -110,7 +118,7 @@ namespace ada {
           // If url’s scheme is "file", then:
           if (url.scheme == "file") {
             // If remaining does not start with "//", validation error.
-            if (pointer + 2 < input.end() && pointer[1] == '/' && pointer[2] == '/') {
+            if (pointer + 2 < pointer_end && pointer[1] == '/' && pointer[2] == '/') {
```

Comment:
Consider using: `std::distance(pointer, pointer_end) < 2`.

Reply:
(Not a request for change.)

Reply:
I didn't know `std::distance` existed. Thanks!

Reply:
Upon research I saw that `std::distance` seems to be slower. https://stackoverflow.com/questions/57258226/stddistance-is-slow-and-how-to-improve-it

Reply:
This seems to be a concern for non-random-access iterators like those used for maps only, though.

Reply:
@addaleax is correct. It should never be slower than pointer arithmetic.